### DISCUSSION TITLE
Add .clang-format

### DISCUSSION THREAD

**Discussion by MarcoBonino** [gem5/gem5#1839] [Q&A] [answered]
Hello!

I would like to have a .clang-format file to format changes in gem5 following some rules...
How people decide how to format their code ? Are there any rules ? Where ?

Thanks!

**Comment by powerjg** [comment, marked as answer]
See #362 The rules are here: https://www.gem5.org/documentation/general_docs/development/coding_style/

I also want clang-format in the source, but as you can see in the discussion, PRs, and other code review, it is incredibly controversial in this community.

**Reply by MarcoBonino** [reply]
Thanks! Hope the PR will get accepted soon!
